src/SFA.DAS.EmployerFinance/Application/Commands/ImportLevyDeclarations/ImportLevyDeclarationsCommandHandler.cs

```diff
@@ -0,0 +1,40 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NServiceBus.UniformSession;
+using SFA.DAS.EmployerFinance.Application.Commands.ImportPayeSchemeLevyDeclarations;
+using SFA.DAS.EmployerFinance.Data;
+
+namespace SFA.DAS.EmployerFinance.Application.Commands.ImportLevyDeclarations
+{
+    public class ImportLevyDeclarationsCommandHandler : AsyncRequestHandler<ImportLevyDeclarationsCommand>
+    {
+        private readonly EmployerFinanceDbContext _db;
+        private readonly IUniformSession _uniformSession;
+
+        public ImportLevyDeclarationsCommandHandler(EmployerFinanceDbContext db, IUniformSession uniformSession)
+        {
+            _db = db;
+            _uniformSession = uniformSession;
+        }
+
+        protected override async Task Handle(ImportLevyDeclarationsCommand request, CancellationToken cancellationToken)
+        {
+            var saga = await _db.LevyDeclarationSagas.SingleAsync(s => s.Id == request.SagaId, cancellationToken);
+            
+            var accountPayeSchemeIds = await _db.AccountPayeSchemes
```

Comment:
Shouldn't we filter out historical AccountPayeSchemes here, i.e. Where(aps => aps.Deleted != null)?

Or use a `HasQueryFilter` in `AccountPayeSchemeConfiguration`?

Reply:
Fixed